qcodes_contrib_drivers/drivers/Vaunix/LDA.py

```diff
@@ -0,0 +1,291 @@
+
+        if sys.platform != "win32":
+            raise OSError(f"LDA is not supported on {sys.platform}.")
+
+        bitness = architecture()[0]
+        if "64bit" in bitness:
+            dll_path = os.path.join(dll_path, "VNX_atten64")
+        elif "32bit" in bitness:
+            dll_path = os.path.join(dll_path, "VNX_atten")
+        else:
+            raise OSError("Unknown bitness of system:", bitness)
+
+        return ctypes.cdll.LoadLibrary(dll_path)
+
+    def get_idn(self) -> Dict[str, Optional[str]]:
+
+        buf = ctypes.create_string_buffer(300)
+        self.dll.fnLDA_GetModelNameA(self.reference, buf)
+        model = str(buf.value.decode())
+
+        return {"vendor": "Vaunix",
+                "model": model,
+                "serial":  self.dll.fnLDA_GetSerialNumber(self.reference),
+                "firmware": self.dll.fnLDA_GetDLLVersion(),
+                }
+
+    def close(self) -> None:
+        self.dll.fnLDA_CloseDevice(self.reference)
+        super().close()
+
+    def save_settings(self) -> None:
+        """
+        Save current settings to memory. Settings are automatically loaded
+        during power on.
+        """
+        self.dll.fnLDA_SaveSettings(self.reference)
+
+
+class LdaChannel(InstrumentChannel):
+    """
+    Channel corresponding to one input-output pair of the LDA digital
+    attenuator.
+    """
+    def __init__(self, parent: Vaunix_LDA,
+                 channel_number: int,
+                 name: str):
+        super().__init__(parent=parent, name=name)
+        self.channel_number = channel_number
+        self.add_parameter("attenuation",
+                           parameter_class=LdaAttenuation,
+                           )
+        wf_vals = LdaWorkingFrequency.get_validator(parent)
+        if wf_vals:
+            self.add_parameter("working_frequency",
+                               parameter_class=LdaWorkingFrequency,
+                               vals=wf_vals,
+                               )
+
+
+class LdaParameter(Parameter):
+    scaling = 1.0  # Scaling from integers from API to physical quantities
+
+    def __init__(self, name: str,
+                 instrument: Union[Vaunix_LDA, LdaChannel],
+                 dll_get_function: Callable, dll_set_function: Callable,
+                 **kwargs):
+        """
+        Parameter associated with one channel of the LDA.
+
+        Args:
+            name: parameter name
+            instrument: parent instrument, either LDA or LDA channel
+            dll_get_function: DLL function that gets the value
+            dll_get_function: DLL function that sets the value
+        """
+        super().__init__(name, instrument, **kwargs)
+        self.dll = instrument.root_instrument.dll
+        self.reference = instrument.root_instrument.reference
+        self.dll_get_function = partial(dll_get_function, self.reference)
+        self.dll_set_function = partial(dll_set_function, self.reference)
```

Comment:
i'd make all of these attributes private in order not to confuse users :)

Reply:
Fixed in a72e36d.